src/frontend/Komponenter/Personoversikt/LagBehandlingModal.tsx

```diff
@@ -0,0 +1,193 @@
+import { useNavigate } from 'react-router-dom';
+import { useToggles } from '../../App/context/TogglesContext';
+import { ToggleName } from '../../App/context/toggles';
+import { EToast } from '../../App/typer/toast';
+import { LagRevurdering } from './LagRevurdering';
+
+export const StyledSelect = styled(Select)`
+    margin-top: 2rem;
+`;
+
+const KnappeWrapper = styled.div`
+    margin-top: 2rem;
+    margin-bottom: 1rem;
+`;
+
+const StyledHovedknapp = styled(Hovedknapp)`
+    margin-right: 1rem;
+`;
+
+interface IProps {
+    visModal: boolean;
+    settVisModal: (bool: boolean) => void;
+    fagsakId: string;
+    settHentTilbakekrevinger: (bool: boolean) => void;
+}
+
+const LagBehandlingModal: React.FunctionComponent<IProps> = ({
+    visModal,
+    settVisModal,
+    fagsakId,
+    settHentTilbakekrevinger,
+}) => {
+    const { toggles } = useToggles();
+    const skalViseValgmulighetForSanksjon = toggles[ToggleName.visValgmulighetForSanksjon];
+    const kanLeggeTilNyeBarnPåRevurdering = toggles[ToggleName.kanLeggeTilNyeBarnPaaRevurdering];
+    const visOpprettTilbakekreving = toggles[ToggleName.visOpprettTilbakekreving];
+    const [feilmeldingModal, settFeilmeldingModal] = useState<string>();
+    const [valgtBehandlingstype, settValgtBehandlingstype] = useState<Behandlingstype>();
+    const [valgtBehandlingsårsak, settValgtBehandlingsårsak] = useState<Behandlingsårsak>();
+    const [valgtDato, settValgtDato] = useState<string>();
+    const [valgtBarn, settValgtBarn] = useState<BarnForRevurdering[]>([]);
+    const [senderInnBehandling, settSenderInnBehandling] = useState<boolean>(false);
+    const kanStarteRevurdering = (): boolean => {
+        return !!(valgtBehandlingstype && valgtBehandlingsårsak && valgtDato);
+    };
+    const { axiosRequest, settToast } = useApp();
+    const navigate = useNavigate();
+
+    const [nyeBarnSidenForrigeBehandling, settNyeBarnSidenForrigeBehandling] = useState<
+        Ressurs<BarnForRevurdering[]>
+    >(byggTomRessurs());
+
+    useEffect(() => {
+        axiosRequest<BarnForRevurdering[], null>({
+            url: `familie-ef-sak/api/behandling/barn/fagsak/${fagsakId}/nye-barn`,
+        }).then((response: RessursSuksess<BarnForRevurdering[]> | RessursFeilet) => {
+            settNyeBarnSidenForrigeBehandling(response);
+        });
+    }, [axiosRequest, fagsakId]);
+
+    const opprettBehandling = () => {
+        settFeilmeldingModal('');
+
+        if (valgtBehandlingstype === Behandlingstype.TILBAKEKREVING) {
+            settSenderInnBehandling(true);
+
+            axiosRequest<Ressurs<void>, null>({
+                method: 'POST',
+                url: `/familie-ef-sak/api/tilbakekreving/fagsak/${fagsakId}/opprett-tilbakekreving`,
+            })
+                .then((response) => {
+                    if (response.status === RessursStatus.SUKSESS) {
+                        settHentTilbakekrevinger(true);
```

Comment:
Vil vi egentlig gjenskape `hentBehandling.rerun()` for `hentTilbakekrevinger` her? Kanskje gjøre tilsvarende som `hentBehandling` ?

Reply:
Er det useRerunnableEffect du tenker på da?

Reply:
Ja, den jeg tenkte på - men den lever jo i behandlingContext, så da blir det kanskje litt annerledes..?